File: app/src/main/java/com/antyzero/smoksmog/job/JobModule.java
package com.antyzero.smoksmog.job;

import android.content.Context;

import com.firebase.jobdispatcher.FirebaseJobDispatcher;
import com.firebase.jobdispatcher.GooglePlayDriver;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
@Singleton
public class JobModule {

    @Provides
    @Singleton
    public FirebaseJobDispatcher provideFirebaseJobDispatcher(Context context) {
        return new FirebaseJobDispatcher(new GooglePlayDriver(context));
    }
}

File: app/src/main/kotlin/com/antyzero/smoksmog/job/JobModule.kt
package com.antyzero.smoksmog.job

import android.content.Context
import com.firebase.jobdispatcher.FirebaseJobDispatcher
import com.firebase.jobdispatcher.GooglePlayDriver
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
@Singleton
class JobModule {

    @Provides
    @Singleton
    internal fun provideFirebaseJobDispatcher(context: Context): FirebaseJobDispatcher {
        return FirebaseJobDispatcher(GooglePlayDriver(context))
    }
}
